src/bidiMapper/domains/script/ScriptProcessor.ts

```diff
@@ -70,11 +74,16 @@ export class ScriptProcessor {
     const preloadScript = new PreloadScript(params, this.#logger);
     this.#preloadScriptStorage.add(preloadScript);
 
-    const cdpTargets = new Set<CdpTarget>(
-      this.#browsingContextStorage
-        .getTopLevelContexts()
-        .map((context) => context.cdpTarget)
-    );
+    const cdpTargets =
+      contexts.size === 0
+        ? new Set<CdpTarget>(
+            this.#browsingContextStorage
+              .getTopLevelContexts()
+              .map((context) => context.cdpTarget)
+          )
+        : new Set<CdpTarget>(
```

Comment:
The same issue applies here. If "contexts" contains the same context twice, then the same cdp target will end up being included twice, which is not desired.

If we do the fix in the first set (`contexts`), then no AI is needed here. It's also possible to do the deduplication here instead. But I think it's cleaner to do it above.